### PR TITLE
SOC-898 New auth flow in wiki language

### PR DESCRIPTION
**Pull request by bkoval**
@rafalkalinski @kvas-damian
https://wikia-inc.atlassian.net/browse/SOC-898

I cannot link the new signup url yet since we don't want this page to be live but we can QA the login path (signup path works in the same way).

### CODE REVIEW

File: front/scripts/main/components/LoginIconComponent.ts [hunk 1]
```diff
@@ -9,7 +9,9 @@ App.LoginIconComponent = Em.Component.extend({
 
 	click: function (): void {
 		if (Mercury.wiki.enableNewAuth) {
-			window.location.href = '/join?redirect=' + encodeURIComponent(window.location.href);
+			window.location.href = '/join?redirect=' +
+			encodeURIComponent(window.location.href) +
+			'&uselang=' + Mercury.wiki.language.content || 'en';
```
Comment (kvas-damian) [inline front/scripts/main/components/LoginIconComponent.ts]: can't we just skip ```uselang``` when there is no language?

File: server/lib/AuthUtils.ts [hunk 2]
```diff
@@ -29,6 +29,14 @@ export function getLoginUrlFromRedirect(redirect: string): string {
 	return url.format(forgotPasswordUrlObj);
 }
 
+export function getNewLoginUrl(request: Hapi.Request): string {
```
Comment (kvas-damian) [inline server/lib/AuthUtils.ts]: I don't like ```new``` in method names. How would we name this method when newer login comes?
Can't it be just ```getLoginUrl```?

Comment (rafalkalinski) [inline server/lib/AuthUtils.ts]: good point!

File: front/scripts/main/components/LoginIconComponent.ts (hunk 1 above)
Comment (rafalkalinski) [inline front/scripts/main/components/LoginIconComponent.ts]: and it would be good if the uselang value will be uri-encoded :)

**Comment by kvas-damian** [comment]
LGTM

**Comment by kvas-damian** [comment]
@bkoval Please resolve conflicts.

BTW I don't see where you pass language to Helios

**Comment by kvas-damian** [comment]
LGTM

**Comment by bkoval** [comment]
Closing to file a new one, trageted at a different branch.

**Pull request closed without merging**